src/Domain.php

```diff
@@ -651,15 +691,63 @@ public function withoutLabel(int $key, int ...$keys): self
         }
 
         if ([] === $labels) {
-            return new self();
+            return new self(null, null, ...$this->getIDNAOptions());
         }
 
         $domain = implode('.', array_reverse($labels));
         $psContent = $this->publicSuffix->getContent();
         if (null === $psContent || '.'.$psContent !== substr($domain, - strlen($psContent) - 1)) {
-            return new self($domain);
+            return new self($domain, null, ...$this->getIDNAOptions());
         }
 
-        return new self($domain, $this->publicSuffix);
+        return new self($domain, $this->publicSuffix, ...$this->getIDNAOptions());
+    }
+    
+    /**
+     * @return array
+     */
+    public function getIDNAOptions(): array
+    {
+        return [$this->asciiIDNAOption, $this->unicodeIDNAOption];
+    }
+    
+    public function getAsciiIDNAOption(): int
+    {
+        return $this->asciiIDNAOption;
+    }
+    
+    public function getUnicodeIDNAOption(): int
+    {
+        return $this->unicodeIDNAOption;
+    }
+    /**
+     * Set IDNA_* options for functions idn_to_ascii, idn_to_utf.
+     * @see https://www.php.net/manual/en/intl.constants.php
+     * @param  int   $forAscii
+     * @param  int   $forUnicode
+     * @return $this
+     */
+    public function setIDNAOptions(int $forAscii, int $forUnicode)
+    {
+        $this->asciiIDNAOption = $forAscii;
+        $this->unicodeIDNAOption = $forUnicode;
+        return $this;
+    }
+    
+    /**
+     * return true if domain contains deviation characters.
+     * @see http://unicode.org/reports/tr46/#Transition_Considerations
+     * @return bool
+     **/
+    public function isTransitionalDifferent(): bool
```

Comment:
Isn't this property accessible in the `$idna_info` variable on convertion why using a `try catch` construct  :thinking: